casper/src/main/resources/PoS.rho

```diff
@@ -1,181 +1,176 @@
-                    if (bondAmount < minimumBond) {
-                      return!((false, "Bond less than minimum!"))
-                    } else if (bondAmount > maximumBond) {
-                      return!((false, "Bond greater than maximum!"))
-                    } else {
-                      return!((true, bondAmount))
-                    }
-                  }
-
-                  // TODO: figure out rate-limiting of bonding
-                  | contract validateBondingRate (@bondAmount, return) = {
-                    return!((true, bondAmount))
-                  }
-
-                  | contract validatePublicKey (@bondAmount, return) = {
-                    new isBondedCh, parametersCh in {
-                      getParameters!(*parametersCh)
-                      | for (_, _, @userPk, _ <- parametersCh) {
-                        isBonded!(userPk, *isBondedCh)
-                        | for (@isBondedResult <- isBondedCh) {
-                          if (isBondedResult) {
-                            return!((false, "Public key ${pk} already bonded." %% {"pk" : userPk}))
-                          } else return!((true, bondAmount))
+/*
+The table below describes the required computations and their dependencies
+
+No. | Dependency | Computation method | Result
+----+------------+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------
+1.  |            | given              | sk = d70491c21a5ba44ffb60a97d9bfbf3f3839b997bde107a3a8ab6872594282740
+2.  |            | given              | timestamp = 1553862086627
+3.  |            | lastNonce          | nonce = 9223372036854775807
+4.  | 1,         | ed25519            | pk = cc87bf7747a8c176714b417ca14a63897b07572876c5e38a7896b6007738ef81
+5.  | 4, 2,      | genIds             | uname = Unforgeable(0xf974f799067b34be1acb7b565e0b160edb23df3e3ad58f200af183ae48b458cd)
+6.  | 3, 5,      | registry           | value = (9223372036854775807, bundle+ {   Unforgeable(0xf974f799067b34be1acb7b565e0b160edb23df3e3ad58f200af183ae48b458cd) })
+7.  | 6,         | protobuf           | toSign = 2a3eaa013b0a0d2a0b10feffffffffffffffff010a2a5a280a243a220a20f974f799067b34be1acb7b565e0b160edb23df3e3ad58f200af183ae48b458cd1001
+8.  | 7, 1,      | ed25519            | sig = 7ff160accda7f76f14f2ae132aaae140ac22b2671e561a58a566bff1a01e49cf372201a5df678aba5ff09171b6af966906deffbf8933efeeddb69a4c204b0503
+9.  | 4,         | registry           | uri = rho:id:cnec3pa8prp4out3yc8facon6grm3xbsotpd4ckjfx8ghuw77xadzt
+----+------------+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------
+*/
+
+new MakePoS,
+    registryInsert(`rho:registry:insertSigned:ed25519`),
+    registryLookup(`rho:registry:lookup`),
+    getParameters(`rho:deploy:params`),
```

Comment:
`getParameters` is only used in `getUser`. Please introduce it in with `parametersCh`.